docs/utilities/config_annotations.md

```diff
@@ -0,0 +1,234 @@
+Configuration Annotations
+=========================
+
+The 1.12.x version of Minecraft Forge introduced novel mod configuration system based on annotations. This new system relies on Forge ASM as to automatically discover mod configuration classes and their respective configurable fields as to make working with configurations as easy as possible. Ideally this new system will encourage the standard implementation of mod configurations as to afford end-users with maximally configurable modded Minecraft experiences.
```

Comment:
This last sentence is weird. We've had configs forever, and either way speculation isn't good to have in docs.